examples/logs-agent/main.tf

```diff
@@ -0,0 +1,178 @@
+  name                     = "${var.prefix}-subnet"
+  vpc                      = ibm_is_vpc.example_vpc.id
+  zone                     = "${var.region}-1"
+  total_ipv4_address_count = 256
+  resource_group           = module.resource_group.resource_group_id
+}
+
+resource "ibm_resource_instance" "cos_instance" {
+  count             = var.is_openshift ? 1 : 0
+  name              = "${var.prefix}-cos"
+  service           = "cloud-object-storage"
+  plan              = "standard"
+  location          = "global"
+  resource_group_id = module.resource_group.resource_group_id
+  tags              = var.resource_tags
+}
+
+# Lookup the current default kube version
+data "ibm_container_cluster_versions" "cluster_versions" {}
+locals {
+  default_version = var.is_openshift ? "${data.ibm_container_cluster_versions.cluster_versions.default_openshift_version}_openshift" : data.ibm_container_cluster_versions.cluster_versions.default_kube_version
+}
+
+resource "ibm_container_vpc_cluster" "cluster" {
+  name                 = var.prefix
+  vpc_id               = ibm_is_vpc.example_vpc.id
+  kube_version         = local.default_version
+  flavor               = "bx2.4x16"
+  worker_count         = "2"
+  entitlement          = var.is_openshift ? "cloud_pak" : null
+  cos_instance_crn     = var.is_openshift ? ibm_resource_instance.cos_instance[0].id : null
+  force_delete_storage = true
+  wait_till            = "Normal"
+  zones {
+    subnet_id = ibm_is_subnet.testacc_subnet.id
+    name      = "${var.region}-1"
+  }
+  resource_group_id = module.resource_group.resource_group_id
+  tags              = var.resource_tags
+}
+
+data "ibm_container_cluster_config" "cluster_config" {
+  cluster_name_id   = ibm_container_vpc_cluster.cluster.id
+  resource_group_id = module.resource_group.resource_group_id
+}
+
+# Sleep to allow RBAC sync on cluster
+resource "time_sleep" "wait_operators" {
+  depends_on      = [data.ibm_container_cluster_config.cluster_config]
+  create_duration = "5s"
+}
+
+##############################################################################
+# Observability Instance
+##############################################################################
+
+
+module "observability_instance" {
+  source  = "terraform-ibm-modules/observability-instances/ibm"
+  version = "2.16.0"
+  providers = {
+    logdna.at = logdna.at
+    logdna.ld = logdna.ld
+  }
+  region                     = var.region
+  log_analysis_provision     = false
+  cloud_monitoring_provision = false
+  activity_tracker_provision = false
+  cloud_logs_instance_name   = "${var.prefix}-cloud-logs"
+  resource_group_id          = module.resource_group.resource_group_id
+  cloud_logs_plan            = "standard"
+  cloud_logs_tags            = var.resource_tags
+  cloud_logs_access_tags     = var.access_tags
+}
+
+data "ibm_is_security_groups" "vpc_security_groups" {
+  depends_on = [ibm_container_vpc_cluster.cluster]
+  vpc_id     = ibm_is_vpc.example_vpc.id
+}
+
```

Comment:
Can you leave a code comment explaining that below code creates a VPE for Cloud logs in the provisioned VPC which allows the agents to access the private Cloud Logs Ingress endpoint